### PR TITLE
Switch Qt6 for Mixxx 2.5

### PR DESCRIPTION
**Pull request by daschuer**
The mac arm64 cross-compile is disabled here, because in Qt6 a full host install is required which exceeds the GitHub size limitations. It will be learnable in the smaller 2.5-rel where Debug builds are not included. 

I have prepared it here https://github.com/daschuer/vcpkg/tree/qt6-rel for testing. 
Mixxx Qt6 builds can be found here: 
https://github.com/daschuer/mixxx/actions/runs/5241045427

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -14,24 +14,15 @@ jobs:
             vcpkg_triplet: x64-windows
             vcpkg_host_triplet: x64-windows           
             vcpkg_overlay_ports: overlay/windows;overlay/ports
-            vcpkg_packages_extras: angle libid3tag libmad qt5-winextras
+            vcpkg_packages_extras: libid3tag libmad
```
Comment (JoergAtGithub) [inline .github/workflows/build.yml:17]: The zip file seems to contain the Angle files libGLESv2.dll, libGLESv2.pdb. Can it be that some cache wasn't cleaned properly?

**Review by JoergAtGithub — CHANGES_REQUESTED** [review]
The zip file contain the Angle files, while the ang

File: .github/workflows/build.yml (hunk 1 above)
Comment (daschuer) [reply .github/workflows/build.yml:17]: I have now deleted the cache and started over.

**Comment by JoergAtGithub** [comment]
The rebuilt removed a lot of files from the Windows Zip-file and reduced the size of the buildenv accordingly:
- Build with active github Actions cache: **2.35**GByte
- Build with manually cleared Github Actions cache: **1.33**GByte
- Build of last Qt5 buildenv: **1.17**GByte

But the size of the Qt6 MacOS Intel buildenv is still huge compared to the Qt5 version:

- Last Github Actions build of this PR: **2.24**GByte
- Build of last Qt5 buildenv: **1.44**GByte

@daschuer Did you also cleared the MacOS cache before rebuild?

I wonder if the cache is working correct at all. I think such cache malfunctionality is pretty dangerous - maybe we should better disable the caching for mixxxdj/VCPKG ?

**Comment by daschuer** [comment]
I cannot spot extra files that are in the package. But the debug information are significant bigger for Qt6 and there are a bunch of extra debug files for QML: 
grafik

It is expected that the current mixxx main branch does not work with this environment. There are some adjustments required I have put into this branch: 
https://github.com/daschuer/mixxx/tree/vcpkg_update_direct_qt6_1015

Do you see the coverarts with the CI build linked above?

**Comment by JoergAtGithub** [comment]
> Do you see the coverarts with the CI build linked above?

There is no Windows build linked.

**Review by JoergAtGithub — APPROVED** [review]
I think this buildenv is Ok. But before it can be used in Main, further changes in mixxxdj/mixxx are needed.
I merge this now, to make it possible to continue with this work.
Thank you!

**Pull request merged**